src/notebooksyncagent.cpp

```diff
@@ -474,7 +474,7 @@ void NotebookSyncAgent::processETags(const QString &uri)
                    report->receivedCalendarResources()) {
             if (!resource.href.contains(mRemoteCalendarPath)) {
                 qCWarning(lcCalDav) << "href does not contain server path:" << resource.href << ":" << mRemoteCalendarPath;
-                mFailingUpdates.insert(uri);
+                mFailingUpdates.insert(uri, QByteArray("Mismatch in hrefs from server response."));
```

Comment:
Is the error data supposed to be human-readable (since I guess it will be displayed in the sync log, eventually)?  If so, should it be some translation id rather than a latin1 string?  Also, in that case, we should try to ensure that the message provides the user with a clear resolution action (e.g. what can they do to resolve the issue).

Reply:
Yes in the long term it should be visible to the user (but quite hidden, mainly accessible for bug reporting). But, like server answers, it doesn't matter if it's in English.

Here, I just created the possibility to gather errors in the logs. I'm not definitive on how these errors should look like. I need them though to investigate how to expose them in the UI. I can tune the messages later.

About the fact that it should provide guidance. I fully agree. Depending how this is done in the UI, I'm thinking on providing tools to actually silent errors on next sync:
- for instance an event that cannot be pushed recurrently to the server could be silenced by the addition of a flag on it.
- …
But this is for later in my opinion, when we have a clearer idea on what to expose to the user.